src/App.tsx

```diff
@@ -15,14 +15,43 @@ import {
 import { AlphaBanner, GweiBar } from '@components/composite';
 import Box from '@mui/material/Box';
 import { useEffect } from 'react';
+import {
+  deleteReferrer,
+  isRefererStored,
+  isValidReferrerStored,
+  isValidReferrerValue,
+  setReferrer,
+} from './utilities/referrer-store/referrer-store';
 
 const App = () => {
+  const [searchParams] = useSearchParams();
+  const searchParamsReferrer = searchParams.get('invitedBy');
+
   useEffect(() => {
     if (process.env.REACT_APP_GTM_CODE) {
       TagManager.initialize({ gtmId: process.env.REACT_APP_GTM_CODE });
     }
   }, []);
 
+  // referrer logic - run everytime params change for invitedBy
+  useEffect(() => {
+    // Earlier, pre-release keys may have used more than 8 characters, but we overwrite these
+    if (!isValidReferrerStored()) {
+      deleteReferrer();
+    }
+
+    const isValidReferrer = isValidReferrerValue(searchParamsReferrer);
+    if (!isValidReferrer) {
+      return;
+    }
+
+    // Referrer is set in URL and has not already been saved in storage; save to storage now
+    // We do deliberately do not overwrite any existing value because we want to credit the first referral link that was followed
+    if (!isRefererStored()) {
```

Comment:
Do we need isReferrerStored or would a call to isValidReferrerStored suffice here?

Reply:
isValidReferrerStored will work as well but isValidReferrerStored = getReferrer + isValidReferrer = 2 fn calls (this is 1).
since this happens after isValidReferrerStored which is on line 39, if there is no valid referrer in localstorage we would delete it, so this will and is more optimal imho.

Reply:
Ok. If you don't make the other change then you don't need to change this. (But if you do you do. :)